### ISSUE TITLE
Modifying kwargs

### ISSUE THREAD

**Issue by yha** [oxinabox/LoggingExtras.jl#18]
This may be a documentation issue, but it seems like there is no easy way to add custom keys to a log event. I initially expected to be able to simply do something like
```
TransformerLogger(current_logger()) do log
    merge(log, (; newkey="value"))
end
```
but that doesn't work, because apparently `TransformerLogger` should return a named tuple with specific fields, and the custom key-value pairs are in a `kwargs` field. Writing to that field also doesn't work because that would change its type.
After digging around the package code, the easiest solution I came up with was calling `LoggingExtras.handle_message_args` (which is not mentioned in the docs), and using to the replace the event with a new one:
```
function merge_log_event(event; kwargs...)
    args = Tuple(event)[1:end-1]   # to drop the kwargs field
    LoggingExtras.handle_message_args(args...; kwargs..., event.kwargs...)
end

logger = TransformerLogger(current_logger()) do event
    merge_log_event(event; newkey="value")
end
```
Maybe something like this should be provided in the package?

**Comment by oxinabox** [comment]
>  the custom key-value pairs are in a kwargs field.

Yes, so what you need to do is access that field to get the old kwargs,
and then use that to build new kwargs,
which you then need to use to build a new log message.

We can't put them at the top-level, with the standard parts of the log message,
as that would disallow working with custom kwargs with names like `msg`,
since that would clash with actual message.

Following is an example of how to do this.
A PR to add something similar to the readme would be appreciated.

```
using Logging, LoggingExtras, Dates 

const date_format = "yyyy-mm-dd HH:MM:SS"

timestamp_kwlogger(logger) = TransformerLogger(logger) do orig_log
  new_kwargs = (timestamp=now(), orig_log.kwargs...)
  new_log = merge(orig_log, (kwargs=new_kwargs,))
  return new_log
end

ConsoleLogger(stdout, Logging.Debug) |> timestamp_kwlogger |> global_logger
```

### Demo
```
julia> @info "Hi"
┌ Info: Hi
└   timestamp = 2019-09-23T11:02:19.403
```
